napari/_app_model/constants/_menus.py

```diff
@@ -41,7 +52,20 @@ class LAYERLIST_CONTEXT:
 
 
 # TODO: add these to docs, with a lookup for what each menu is/does.
-_CONTRIBUTABLES = {MenuId.LAYERLIST_CONTEXT.value}
+_CONTRIBUTABLES = {
+    MenuId.LAYERLIST_CONTEXT.value,
+    MenuId.LAYERS_CONVERT_DTYPE.value,
+    MenuId.LAYERS_PROJECT.value,
+    MenuId.TOOLS_ACQUISITION.value,
+    MenuId.TOOLS_CLASSIFICATION.value,
+    MenuId.TOOLS_FILTERS.value,
+    MenuId.TOOLS_MEASUREMENT.value,
+    MenuId.TOOLS_SEGMENTATION.value,
+    MenuId.TOOLS_PROJECTION.value,
+    MenuId.TOOLS_TRANSFORM.value,
+    MenuId.TOOLS_UTILITIES.value,
+    MenuId.TOOLS_VISUALIZATION.value,
+}
```

Comment:
does this set need to be manually maintained, or maybe could be automatically generated from `MenuId.__members__`?

Reply:
This needs to be manually managed. The contributable ones are not the full list of all menus

Reply:
I still prefer to add some class methods to Enum, instead of a separate positions. It will be one place in the context of application logic.

Reply:
I tried adding the contributable as a boolean to the enum and some class methods to check if a menu id is contributable or not. My biggest complaint is that we have to either use the private member map in the enum class or loop through all values to match the id in question. Either way the code doesn't look very nice. If you still prefer this approach, I can push what I have and see if you like them.

Reply:
I don't super mind these being declared manually in each spot, I do think it's part of the necessary curation work that we'll need to do as we update these menus. Happy to be challenged on that.

I think if we're going to go with this whitelisting approach, we need a warning [here](https://github.com/napari/napari/blob/9ea0159ad2b690556fe56ce480886d8f0b79ffae/napari/plugins/_npe2.py#L353) that asks users to suggest that menu for whitelisting, and mentions the menus that are being skipped.

Reply:
> I don't super mind these being declared manually in each spot, I do think it's part of the necessary curation work that we'll need to do as we update these menus. Happy to be challenged on that.

I also don't mind if there isn't a straightforward way to generate this list, but it might be nice to have a test to make sure these items are consistent with `MenuId`. Is that possible?

> I think if we're going to go with this whitelisting approach, we need a warning [here](https://github.com/napari/napari/blob/9ea0159ad2b690556fe56ce480886d8f0b79ffae/napari/plugins/_npe2.py#L353) that asks users to suggest that menu for whitelisting, and mentions the menus that are being skipped.

agreed!

Reply:
> I also don't mind if there isn't a straightforward way to generate this list, but it might be nice to have a test to make sure these items are consistent with `MenuId`. Is that possible?

The most common approach that I see is to define a class method for each category in enum to return a list of enum members belonging to a given category. Then you store such information in class, not in other files.

Reply:
> > I also don't mind if there isn't a straightforward way to generate this list, but it might be nice to have a test to make sure these items are consistent with `MenuId`. Is that possible?
> 
> The most common approach that I see is to define a class method for each category in enum to return a list of enum members belonging to a given category. Then you store such information in class, not in other files.

That makes sense! I probably took auto-generation too literally. I can move the contributable list to the menuID and make it a class method